test/unit/transformer-oracle.spec.ts

```diff
@@ -192,4 +219,57 @@ describe('TransformerOracle', () => {
       interface: IERC20__factory.createInterface(),
     });
   });
+
+  function executeRedirectTest<T extends keyof ITokenPriceOracle['functions']>({
+    returns,
+    func,
+    params,
+  }: {
+    returns?: any;
+    params: (mappedTokenA: string, mappedTokenB: string) => Parameters<ITokenPriceOracle['functions'][T]>;
+    func: T;
+  }) {
+    describe(func, () => {
+      executeTest({
+        when: 'mapped are tokenA and tokenB',
+        mappedA: TOKEN_A,
+        mappedB: TOKEN_B,
+      });
+      executeTest({
+        when: 'mapped are tokenA and underlyingTokenB',
+        mappedA: TOKEN_A,
+        mappedB: UNDERLYING_TOKEN_B,
+      });
+      executeTest({
+        when: 'mapped are underlyingTokenA and tokenB',
+        mappedA: UNDERLYING_TOKEN_A,
+        mappedB: TOKEN_B,
+      });
+      executeTest({
+        when: 'mapped are underlyingTokenA and underlyingTokenB',
+        mappedA: UNDERLYING_TOKEN_A,
+        mappedB: UNDERLYING_TOKEN_B,
+      });
+    });
+    function executeTest({ when: title, mappedA, mappedB }: { when: string; mappedA: string; mappedB: string }) {
+      when(title, () => {
+        let returned: any;
+        given(async () => {
+          await transformerOracle.setUnderlying(TOKEN_A, TOKEN_B, mappedA, mappedB);
+          if (returns) {
+            underlyingOracle[func].returns(returns);
+          }
+          returned = await (transformerOracle[func] as any)(...params(TOKEN_A, TOKEN_B));
+        });
```

Comment:
Idea of mocking internal mapping here is to avoid externalities / using the registries mock, right?

Reply:
`mapPairToUnderlying` does a few things other than calling the registry, and in a few more PRs down the line, it will get even more complicated. The idea here is to just test that the function uses whatever `mapPairToUnderlying` returns